app/controllers/course/condition/achievements_controller.rb

```diff
@@ -1,13 +1,17 @@
 class Course::Condition::AchievementsController < Course::ConditionsController
-  load_and_authorize_resource :achievement_condition,
-                              class: Course::Condition::Achievement.name, parent: false
+  load_resource :achievement_condition, class: Course::Condition::Achievement.name, parent: false
+  authorize_resource :achievement_condition, class: Course::Condition::Achievement.name,
```

Comment:
you can add a before_action to set the `@achievement_condition.course` before authorize_resource, then you won't need to manually authorize.